src/umpire/strategy/FixedPool.cpp

```diff
@@ -105,7 +105,9 @@ FixedPool::allocInPool(Pool& p) noexcept
     }
   }
 
-  UMPIRE_ASSERT("FixedPool: Logic error in allocate");
+  // The method should either return nullptr at the top, or within the
+  // loop. This should never be reached.
+  UMPIRE_ERROR("A logic error occured");
```

Comment:
This can be caught which is not what I think we want.  What can the user do with this?

I really think this should be an assertion and not an error as it points to a bug in the Umpire implementation.